File: prototype/Assets/microcosmicWar/Scripts/DefenseTowerNetView.js

var defenseTower:DefenseTower;
var life:Life;

function Awake()
{
	if(!defenseTower)
		defenseTower=gameObject.GetComponentInChildren(DefenseTower);
	if(!life)
		life=gameObject.GetComponentInChildren(Life);
	
	//print(!zzCreatorUtility.isHost());
	if( !zzCreatorUtility.isHost() )
	{
		Destroy(defenseTower.GetComponentInChildren(AiMachineGunAI));
	}
}


function OnSerializeNetworkView(stream : BitStream, info : NetworkMessageInfo)
{
	life.OnSerializeNetworkView(stream,info);
	
	var lAngle:float;
	var lFire:boolean;
	var lAimAngular:float;
	//var lFace:int;
	
	//---------------------------------------------------
	if (stream.isWriting)
	{
		lAngle=defenseTower.getAngle();
		lFire=defenseTower.inFiring();
		lAimAngular=defenseTower._getSmoothAngle();
	//	lFace=defenseTower.getFaceDirection();
		
	}
	
	//---------------------------------------------------
	stream.Serialize(lAngle);
	stream.Serialize(lFire);
	stream.Serialize(lAimAngular);
	//stream.Serialize(lFace);
	
	//---------------------------------------------------
	if(stream.isReading)
	{
		defenseTower.setAngle(lAngle);
		defenseTower.setFire(lFire);
		defenseTower._setSmoothAngle(lAimAngular);
	//	defenseTower.setFaceDirection(lFace);
		
	}
}

File: prototype/Assets/microcosmicWar/Scripts/DefenseTowerNetView.cs

using UnityEngine;
using System.Collections;

public class DefenseTowerNetView : MonoBehaviour
{


    public DefenseTower defenseTower;
    public Life life;

    void Awake()
    {
        if (!defenseTower)
            defenseTower = gameObject.GetComponentInChildren<DefenseTower>();
        if (!life)
            life = gameObject.GetComponentInChildren<Life>();

        //print(!zzCreatorUtility.isHost());
        if (!zzCreatorUtility.isHost())
        {
            Destroy(defenseTower.GetComponentInChildren<AiMachineGunAI>());
        }
    }


    void OnSerializeNetworkView(BitStream stream, NetworkMessageInfo info)
    {
        life.OnSerializeNetworkView(stream, info);

        float lAngle=0;
        bool lFire=false;
        float lAimAngular=0;
        //int lFace;

        //---------------------------------------------------
        if (stream.isWriting)
        {
            lAngle = defenseTower.getAngle();
            lFire = defenseTower.inFiring();
            lAimAngular = defenseTower._getSmoothAngle();
            //	lFace=defenseTower.getFaceDirection();

        }

        //---------------------------------------------------
        stream.Serialize(ref lAngle);
        stream.Serialize(ref lFire);
        stream.Serialize(ref lAimAngular);
        //stream.Serialize(lFace);

        //---------------------------------------------------
        if (stream.isReading)
        {
            defenseTower.setAngle(lAngle);
            defenseTower.setFire(lFire);
            defenseTower._setSmoothAngle(lAimAngular);
            //	defenseTower.setFaceDirection(lFace);

        }
    }
}